typing/typecore.ml

```diff
@@ -5817,23 +5888,41 @@ and type_cases
      is to typecheck the guards and the cases, and then to check for some
      warnings that can fire in the presence of guards.
   *)
-  map_half_typed_cases category env ty_arg ty_res loc caselist ~check_if_total
+  map_half_typed_cases ?conts category env ty_arg ty_res loc caselist
+    ~check_if_total
     ~type_body:begin
-      fun { pc_guard; pc_rhs } pat ~ext_env ~ty_expected ~ty_infer
+      fun { pc_guard; pc_rhs } pat ~ext_env ~cont ~ty_expected ~ty_infer
           ~contains_gadt:_ ->
+        let cont, ext_env' =
+          match cont with
+          | Some (id, desc) ->
+              let ext_env =
+                Env.add_value ~check:(fun s -> Warnings.Unused_var_strict s)
```

Comment:
This might be already fixed by @Octachron sub-Pr avsm#4, but the unused variable check does not seem to work with named continuations. Typing `try () with effect E, k -> ();;` does not warn that `k` is unused.

Reply:
@Octachron can you confirm?

Reply:
Thank you.